lib/src/component/error_boundary.dart

```diff
@@ -197,6 +207,33 @@ class ErrorBoundaryComponent<T extends ErrorBoundaryProps, S extends ErrorBounda
     ..showFallbackUIOnError = props.fallbackUIRenderer != null
   );
 
+  @mustCallSuper
+  @override
+  void componentWillMount() {
+    super.componentWillMount();
+
+    _logger = Logger(_getLoggerName());
+  }
+
+  @mustCallSuper
+  @override
+  void componentWillUnmount() {
+    super.componentWillUnmount();
+
+    _logger.clearListeners();
+  }
+
+  @mustCallSuper
+  @override
+  void componentWillReceiveProps(Map nextProps) {
+    super.componentWillReceiveProps(nextProps);
+
+    final tNextProps = typedPropsFactory(nextProps);
+    if (tNextProps.loggerName != props.loggerName) {
+      _logger = Logger(_getLoggerName(tNextProps));
```

Comment:
I don't think we should be caching this value. The drawbacks of increased code complexity and number of test cases far outweigh the negligible perf benefits that caching might provide, especially since errors are expected to happen infrequently.

Reply:
I'm not sure I follow what you mean by caching @greglittlefield-wf ?

You're saying don't store the `Logger` instance on a class field?

Reply:
Right; we're caching the value of `_logger`  when we should just compute it when it's needed